src/conf.py

```diff
@@ -1215,33 +1215,87 @@ def makeBanmask(self, hostmask, options=None, channel=None, network=None):
         isn't specified via options, the value of
         conf.supybot.protocols.irc.banmask is used.
 
+        Unlike :meth:`makeExtBanmasks`, this is guaranteed to return an
+        RFC1459-like mask, suitable for ircdb's ignore lists.
+
         options - A list specifying which parts of the hostmask should
         explicitly be matched: nick, user, host.  If 'exact' is given, then
-        only the exact hostmask will be used."""
-        if not channel:
-            channel = dynamic.channel
+        only the exact hostmask will be used.
+        """
         if not network:
             network = dynamic.irc.network
+        if not options:
+            options = supybot.protocols.irc.banmask.getSpecific(
+                network, channel)()
+        options = [option for option in options if option != 'account']
+        print(hostmask, options)
+        masks = self.makeExtBanmasks(
+            hostmask, options, channel, network=network)
+        assert len(masks) == 1, 'Unexpected number of banmasks: %r' % masks
+        print(masks)
+        return masks[0]
+
+    def makeExtBanmasks(self, hostmask, options=None, channel=None, *, network):
+        """Create banmasks from the given hostmask.  If a style of banmask
+        isn't specified via options, the value of
+        conf.supybot.protocols.irc.banmask is used.
+
+        Depending on the options and configuration, this may return a mask
+        in the format of an extban (eg. "~account:foobar" on UnrealIRCd).
+        If this is unwanted (eg. to pass to ircdb's ignore lists, use
+        :meth:`makeBanmask` instead)
+
+        options - A list specifying which parts of the hostmask should
+        explicitly be matched: nick, user, host.  If 'exact' is given, then
+        only the exact hostmask will be used.
+        If 'account' is given (and not after 'exact') and the user is
+        logged in and the server supports account extbans, then an account
+        extban is returned instead.
+        """
+        if not channel:
+            channel = dynamic.channel
         (nick, user, host) = ircutils.splitHostmask(hostmask)
         bnick = '*'
         buser = '*'
         bhost = '*'
         if not options:
             options = supybot.protocols.irc.banmask.getSpecific(
                 network, channel)()
+
+        add_star_mask = False
+        masks = []
+
         for option in options:
             if option == 'nick':
```

Comment:
I usually prefer Enum types instead of fixed string inputs, especially if we silently ignore invalid options.

Reply:
ditto

Reply:
I guess I can allow both and deprecate the string though.

Reply:
Heh, I'll just log invalid options instead.